charts/sourcegraph-appliance/templates/role.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: sg-appliance-role
+  labels:
+    category: rbac
+    deploy: sourcegraph
+    app.kubernetes.io/component: appliance
+rules:
```

Comment:
When I deployed this, the reconcile loop got stuck and not all sourgraph resources were able to be provisioned. With the following changes, I was able to overcome this:

I added the following resources:

```
      - roles
      - rolebindings
      - events
      - endpoints
      - ingresses
```

Which necessitated adding the following apiGroups:

```
      - "rbac.authorization.k8s.io"
      - "networking.k8s.io"
```

Note that ingresses was only needed because the appliance was attempting to ensure that frontend's ingress was deleted (!), because the configmap doesn't declare that to be enabled. This is the sort of thing we'll want to align the helm-install, the appliance frontend resource provisioning behaviour, and self-update on in that issue you created (thanks!). So we might want to take ingresses and its apigroup back out at some point - I'm really not sure.

Note that events is only needed so that the appliance can create events on its own configmap, which it does in the reconcile loop.

We also need the following verbs:

```
      - delete
      - patch
      - update
```

Being able to run this not-in-god-mode (with a local kubeconfig pointing to a test cluster) exposed a typo in the prometheus rbac. Fix is here: https://github.com/sourcegraph/sourcegraph/pull/63540

With all that, I was able to get the reconcile loop finished, but with frontend initContainers crashlooping because they don't contain the merged DEPLOY_TYPE fix in our release branches yet 🙏